From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Niklas Haas <git@haasn.xyz>
Date: Sun, 21 Mar 2021 17:18:20 +0100
Subject: [PATCH] vo_gpu: placebo: keep track of texture sample mode

This fixes an issue where dithering was effectively broken on libplacebo
versions >= 103 because the dither texture was being sampled with
edge-clamped rather than repeating semantics.
---
 video/out/placebo/ra_pl.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/video/out/placebo/ra_pl.c b/video/out/placebo/ra_pl.c
index d8e8729..09ff742 100644
--- a/video/out/placebo/ra_pl.c
+++ b/video/out/placebo/ra_pl.c
@@ -189,6 +189,10 @@ static struct ra_tex *tex_create_pl(struct ra *ra,
         return NULL;
     }
 
+    // Keep track of these, so we can correctly bind them later
+    ratex->params.src_repeat = params->src_repeat;
+    ratex->params.src_linear = params->src_linear;
+
     return ratex;
 }
 
